inception-pdf-editor/src/main/java/de/tudarmstadt/ukp/inception/pdfeditor/pdfanno/PdfAnnoPanel.java

```diff
@@ -42,28 +48,55 @@
     private @SpringBean DocumentService documentService;
 
     private AbstractAjaxBehavior pdfProvider;
+
+    private AbstractAjaxBehavior pdftxtProvider;
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
     
     public PdfAnnoPanel(String aId, IModel<AnnotatorState> aModel)
     {
         super(aId, aModel);
 
+        SourceDocument doc = aModel.getObject().getDocument();
+
+        File pdfFile = documentService.getSourceDocumentFile(doc);
```

Comment:
This is probably a problem because the PDF file name is obtained only when the PdfAnnoPanel is first created but not when e.g. switching from one document to another. I'd leave it were it was in `onRequest()` below.

Reply:
So should I get it twice then?
I need it in the second `onRequest()`, for pdfextract.